src/lib/config_parser.c

```diff
@@ -129,6 +129,11 @@ char *singularity_config_get_value(char *key) {
     return(NULL);
 }
 
+char *singularity_config_get_value_default(char *key, const char *def) {
```

Comment:
The problem is that we distribute a _config_ file, not a _defaults_ file.  That means, on upgrade, we have no clue as to whether our updated configs are in place.  For example, on RPM-based platforms, updated configs will be saved as `.rpmnew` if the admin changes _any_ line of the config file.  (We've already seen this occur.)

Hence, we're left with the unenviable task of making sure the code defaults match the config defaults (NOTE: other projects I have worked with solve this by either auto-generating the config file from the code defaults at build-time or simply removing defaults from the config file).

We already do this for all the boolean configuration knobs; this just appears the first time we've done it for strings.

Yes, the configuration system code really needs an uplift to make it future-proof.